### DISCUSSION TITLE
How to properly call onChange?

### DISCUSSION THREAD

**Discussion by OnkelTem** [react-hook-form/react-hook-form#10985] [Q&A]
I have a code like this:

```ts
  const registerProps = register(name)

  return (
      <Select
        {...registerProps}
        onChange={(value) => { ... }}
      />
  )
```  

and I need to call the original `onChange` but when I do like:

```ts
onChange={(value) => { registerProps.onChange({ target: { value } }) }}
```

The value is not set and the form isn't being revalidated. This is very strange because in this case it's not clear how RHF gets values.

So I have to update form values and trigger validation manually:

```ts
onChange={(value) => {
  setValue(name, value)
  trigger(name)
}}
```

but this also doesn't work well, as my Select component doesn't get a value when I pick it for the first time. Later on, however, it begins to work. 

Any ideas how to correctly call original onChange?

**Comment by Moshyfawn** [comment]
```jsx
<Select  {...register(name, { onChange })} />
```

or 

```jsx
const registerProps = register(name, { onChange: (value) => { ... } })

return <Select {...registerProps} />
```

**Reply by OnkelTem** [reply]
Sorry, I don't get it. I need to call the original onChange, while you show how to call a custom onChange. It's basically a reverse task.
